### PR TITLE
Add infrastructure and helpers for skipping diagnose checks

### PR DESCRIPTION
**Pull request by sgmiller**

### CODE REVIEW

**Review by HridoyRoy — APPROVED** [review]

File: vault/diagnose/helpers.go [hunk 1]
```diff
@@ -47,15 +49,39 @@ func New() *Session {
...
+func (s *Session) IsSkipped(skipName string) bool {
+	return s.skip[skipName]
+}
+
 // Context returns a new context with a defined diagnose session
 func Context(ctx context.Context, sess *Session) context.Context {
 	return context.WithValue(ctx, diagnoseSession, sess)
 }
 
+// CurrentSession retrieves the active diagnose section from the context, or nil if none.
```
Comment (swayne275) [inline vault/diagnose/helpers.go:74]:
```suggestion
// CurrentSession retrieves the active diagnose session from the context, or nil if none.
```

File: vault/diagnose/helpers.go [hunk 2]
```diff
@@ -154,7 +184,7 @@ func Test(ctx context.Context, spanName string, function func(context.Context) e
 // complete within the timeout, e.g.
 //
 // diagnose.Test(ctx, "my-span", diagnose.WithTimeout(5 * time.Second, myTestFunc))
-func WithTimeout(d time.Duration, f func(context.Context) error) func(ctx context.Context) error {
+func WithTimeout(d time.Duration, f func(context.Context) error) func(context.Context) error {
```
Comment (swayne275) [inline vault/diagnose/helpers.go:187]: might be helpful to the reader to define a type for `func(context.Context) error`?

File: vault/diagnose/helpers_test.go [hunk 3]
```diff
@@ -70,6 +75,7 @@ func makeCoffee(ctx context.Context) error {
 
 	SpotCheck(ctx, "pick-scone", pickScone)
 
+	Test(ctx, "dispose-grounds", Skippable("dispose-grounds", disposeGrounds))
```
Comment (swayne275) [inline vault/diagnose/helpers_test.go:78]: just to verify, Test does the appropriate thing if given a function that returns `nil`?

**Review by swayne275 — APPROVED** [review]
looks good! approving but have a few comments/questions

File: vault/diagnose/helpers.go (hunk 2 above)
Comment (sgmiller) [reply vault/diagnose/helpers.go:187]: For code cleanliness, yeah, good idea.

File: vault/diagnose/helpers_test.go (hunk 3 above)
Comment (sgmiller) [reply vault/diagnose/helpers_test.go:78]: Yeah, marks the span successful (but will show up as Skipped if the skipped event was added)

**Pull request merged**